sdk/core/core/test/cmocka/test_az_http.c

```diff
@@ -512,6 +521,60 @@ static void test_http_response_header_validation_space(void** state)
   }
 }
 
+static void test_http_response_append(void** state)
+{
+  (void)state;
+  {
+    uint8_t buffer[10];
+    az_http_response response = { 0 };
+    az_span append_this = AZ_SPAN_FROM_STR("0123456789");
+    assert_return_code(az_http_response_init(&response, AZ_SPAN_FROM_BUFFER(buffer)), AZ_OK);
+    assert_return_code(az_http_response_append(&response, append_this), AZ_OK);
+    assert_memory_equal(buffer, az_span_ptr(append_this), 10);
+  }
+}
+
+static void test_http_response_append_overflow(void** state)
+{
+  (void)state;
+  {
+    uint8_t buffer[10] = "..........";
```

Comment:
@vhvb1989 the CI is failing because the buffer is too small to hold the null terminator of the string.

https://dev.azure.com/azure-sdk/public/_build/results?buildId=389807&view=logs&j=d6724728-8c00-5781-d729-5591dee122a6&t=87871560-6589-5018-de6c-e75f1d62f018&l=135
```text
D:\a\1\s\sdk\core\core\test\cmocka\test_az_http.c(541,1): error C2220: the following warning is treated as an error [D:\a\1\s\build\sdk\core\core\test\cmocka\az_core_test.vcxproj]
D:\a\1\s\sdk\core\core\test\cmocka\test_az_http.c(541,1): warning C4295: 'buffer': array is too small to include a terminating null character [D:\a\1\s\build\sdk\core\core\test\cmocka\az_core_test.vcxproj]
D:\a\1\s\sdk\core\core\test\cmocka\test_az_http.c(564,1): warning C4295: 'buffer': array is too small to include a terminating null character [D:\a\1\s\build\sdk\core\core\test\cmocka\az_core_test.vcxproj]
```

Reply:
thank you.
I was forgeting about this PR